packages/utils/telemetry-utils/src/batchedTelemetryHelper.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+type ICustomDataMap = Record<string, number>;
+
+/**
+ * Helper class that accumulates custom data and sends it to the logger every time a specified number (i.e., `threshold`) of logs is reached.
+ */
+export class BatchedTelemetryHelper {
+	disposed: boolean = false;
+
+	// Stores value of the custom data passed into the logger.
+	private readonly customDataMap: Map<string, number> = new Map<string, number>();
+	// Counter to keep track of the number of times the log function is called.
+	private counter = 0;
+
+	/**
+	 * @param eventBase -
```

Comment:
Nit: I would recommend moving these param docs down to each individual property in the constructor as normal tsdoc comments. Intellisense knows to treat these as constructor parameter docs, in addition to property-level docs, but the inverse is not true.

Reply:
Is it okay to use a block comment over inline comment inside the constructor? As far as I know, it is recommended to use inline comment inside the method itself.

Reply:
Properties declared inline as constructor parameters are a special case. Tooling generally supports documenting the properties better than documenting the parameters at the function level. In particular, our API docs end up looking a lot better if you document the properties.

Reply:
But you do want to use a complete `/** TSDoc comment */` on each.